src/LuaBody.cpp

```diff
@@ -236,8 +239,10 @@ static int l_body_is_more_important_than(lua_State *l)
 	else if(a == Object::Type::STAR) result = true;
 	// any (non-star) object is smaller than a star
 	else if(b == Object::Type::STAR) result = false;
-	// a gas giant is larger than anything but a star
-	else if(a_gas_giant) result = true;
+	// a gas giant is larger than anything but a star,
+	// but remember to keep total order in mind: if both are
+	// gas giants, order alphabetically
+	else if(a_gas_giant) result = !b_gas_giant || body->GetLabel() < other->GetLabel();
```

Comment:
Yup, the planet sub/super-typing adds to the complexity of this comparison chain. Just coming in to the code base, it took me a good while to figure out why it was failing.